plugins/woocommerce/src/StoreApi/Utilities/CartController.php

```diff
@@ -486,13 +486,29 @@ public function validate_cart() {
 		}
 	}
 
+	/**
+	 * When placing an order, validate that the cart is not empty.
+	 *
+	 * @throws InvalidCartException Exception if the cart is empty.
+	 */
+	public function validate_cart_not_empty() {
+		$cart_items = $this->get_cart_items();
+
+		if ( empty( $cart_items ) ) {
+			throw new InvalidCartException(
+				'woocommerce_cart_error',
+				new WP_Error( 'woocommerce_rest_cart_empty', __( 'Cannot place an order, your cart is empty.', 'woocommerce' ), 400 ),
+				400
+			);
+		}
+	}
+
 	/**
 	 * Validate all items in the cart and check for errors.
 	 *
 	 * @throws InvalidCartException Exception if invalid data is detected due to insufficient stock levels.
 	 */
 	public function validate_cart_items() {
-		$cart       = $this->get_cart_instance();
```

Comment:
Unused var, and I don't think we need to call get_cart_instance for any other reason here.